scrapy_poet/page_input_providers.py

```diff
@@ -178,7 +178,14 @@ def fingerprint(self, to_provide: Set[Callable], request: Request) -> str:
         return json.dumps(fp_data, ensure_ascii=False, sort_keys=True)
 
     def serialize(self, result: Sequence[Any]) -> Any:
-        return [attr.asdict(response_data) for response_data in result]
+        return [self._to_dict(response_data) for response_data in result]
 
     def deserialize(self, data: Any) -> Sequence[Any]:
-        return [ResponseData(**response_data) for response_data in data]
+        return [HttpResponse(**response_data) for response_data in data]
+
+    @staticmethod
+    def _to_dict(data: Any) -> dict:
+        return attr.asdict(
+            data,
+            filter=lambda attrib, _: not attrib.name.startswith("_")
```

Comment:
Is it filtering out the encoding argument? I'm not sure it'd work properly. For example, if headers suggest cp1252 encoding, but user passes utf-8 explicitly, encoding would be set to utf-8. It seems this won't survive the serialize/deserialize roundtrip.

Reply:
Good point. Updated this on https://github.com/scrapinghub/scrapy-poet/pull/67/commits/64263a9ab64e0d672eae10132f506ff911f6eacd